### PR TITLE
emq-apinf architecture

### PR DESCRIPTION
**Pull request by frenchbread**
Closes https://github.com/apinf/platform/issues/2560

### CODE REVIEW

**Comment by bajiat** [comment]
@phanimahesh and @marla-singer Can you also take a look at the documentation and see if it covers the necessary information to continue with development / integration?

**Comment by phanimahesh** [comment]
The diagram is accurate.

@frenchbread, can you also add the table schemas, and a short note on how updates propogate from apinf to emq? (autoform hooks, then rest api, then to db, and emq queries db).

**Comment by frenchbread** [comment]
@phanimahesh Good point, thanks. Will do

File: docs/develop/emq/emq-apinf-architecture.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+     R |                                               |
+     O |  HTTP  +--------------+                       |
+     X +------->| API Umbrella |                       |
+     I |        +--------------+                       |
+     E |                                               |
+     S |                                               |
+       |                                               V
+       |  MQTT  +--------------+                +--------------+
+       +------->| EMQ (EMQTTD) |                | EMQ-REST-API |
+                +--------------+                +--------------+
```
Comment (sa1) [inline docs/develop/emq/emq-apinf-architecture.md]: There is no data flow occuring from APInf to EMQ via MQTT, right? As far as I understand, APInf talks with EMQ only through EMQ-REST-API. The arrows here are a bit confusing, in that case.

Comment (frenchbread) [reply docs/develop/emq/emq-apinf-architecture.md]: > As far as I understand, APInf talks with EMQ only through EMQ-REST-API.

@sa1 That's right. The on the left side there is a list of proxies that are attached to APInf. As there is no MQTT flow between APInf & EMQ, EMQ is still attached to APInf the similar way as API Umbrella. I'll update this, so it's more clear. Thanks.

**Comment by frenchbread** [comment]
Please review.

**Comment by phanimahesh** [comment]
There's detail but it is different from what I expected. Let's take someone else's opinion. @sa1 and @marla-singer, any comments?

**Comment by sa1** [comment]
Docs seem okay to me.
Are there concerns on whether the mongodb collection and the postgres db remain consistent with each other? If those are addressed, we could have some notes on them or on why its not a concern.

**Comment by phanimahesh** [comment]
Current sync is implemented ad hoc. No formal proofs or designs were
attempted to provide consistency guarantees between datastores.

It works for now. Sometime in the future, I would like to overhaul it to
not store anything in mongo at all. Currently damir did it this way since
it is the pattern everywhere else.

I see no point in storing users, clients and acls twice. There are three
options that can get rid of potential data inconsistency issues: using
mongo with emq directly, asking emq to query apinf via http, or to make all
data bypass mongo and go to postgres. I'm inclined towards the third for
various reasons.

Ideally mongo should have only what is needed to render catalogue. All
other data must be in individual proxies and should be managed via apis
directly without making copies. But it isn't that straightforward either.
We'll handle it later when I get some free time on my hands.

Long story short, currently we try to keep things in sync. We haven't
analysed it formally for potential issues. And sync is one way from mongo
to postgres only.

**Comment by frenchbread** [comment]
FYI, the current `emq-users` management implementation users postgres directly, so no sync, or any mongodb usage there.

Next, is's possible to follow that pattern for `emq-acls`. With addition of drag-and-drop feature to re-order items in the list.

**Comment by marla-singer** [comment]
@frenchbread Everything is okay for me

**Comment by phanimahesh** [comment]
LGTM from me then

**Comment by bajiat** [comment]
@brylie Would you have time merge this documentation PR?
